core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -876,7 +876,27 @@ class DAGScheduler(
     // serializable. If tasks are not serializable, a SparkListenerStageCompleted event
     // will be posted, which should always come after a corresponding SparkListenerStageSubmitted
     // event.
-    stage.makeNewStageAttempt(partitionsToCompute.size)
+    val taskIdToLocations = try {
+      val localities = stage match {
```

Comment:
Can you move the initialization of the stage attempt to outside of (and after) the try/catch? It's easy to miss as-is.